WordPressKit/RemoteVideoPressVideo.swift

```diff
@@ -0,0 +1,128 @@
+
+    enum CodingKeys: String, CodingKey {
+        case id, title, description, width, height, duration, displayEmbed, allowDownload, rating, privacySetting, posterURL, originalURL, watermarkURL, bgColor, blogId, postId, finished, token
+    }
+
+    public init(dictionary metadataDict: NSDictionary, id: String) {
+        self.id = id
+
+        title = metadataDict.string(forKey: "title")
+        videoDescription = metadataDict.string(forKey: "description")
+        width = metadataDict.number(forKey: "width")?.intValue
+        height = metadataDict.number(forKey: "height")?.intValue
+        duration = metadataDict.number(forKey: "duration")?.intValue
+        displayEmbed = metadataDict.object(forKey: "display_embed") as? Bool
+        allowDownload = metadataDict.object(forKey: "allow_download") as? Bool
+        rating = metadataDict.string(forKey: "rating")
+        if let privacySettingValue = metadataDict.number(forKey: "privacy_setting")?.intValue, let privacySettingEnum = VideoPressPrivacySetting.init(rawValue: privacySettingValue) {
+            privacySetting = privacySettingEnum
+        }
+        if let poster = metadataDict.string(forKey: "poster") {
+            posterURL = URL(string: poster)
+        }
+        if let original = metadataDict.string(forKey: "original") {
+            originalURL = URL(string: original)
+        }
+        if let watermark = metadataDict.string(forKey: "watermark") {
+            watermarkURL = URL(string: watermark)
+        }
+        bgColor = metadataDict.string(forKey: "bg_color")
+        blogId = metadataDict.number(forKey: "blog_id")?.intValue
+        postId = metadataDict.number(forKey: "post_id")?.intValue
+        finished = metadataDict.object(forKey: "finished") as? Bool
+    }
+
+    /// Returns the specified URL adding the token as a query parameter, which is required to play private videos.
+    /// - Parameters:
+    ///   - url: URL to include the token.
+    ///
+    /// - Returns: The specified URL with the token as a query parameter.
+    @objc(getURLWithToken:)
+    public func getURLWithToken(url: URL?) -> URL? {
+        guard var urlWithToken = url else {
+            return nil
+        }
+        if let token = self.token, var urlComponents = URLComponents(url: urlWithToken, resolvingAgainstBaseURL: true) {
+            let metadataTokenParam = URLQueryItem(name: "metadata_token", value: token)
+            var queryItems: [URLQueryItem] = urlComponents.queryItems ?? []
+            queryItems.append(metadataTokenParam)
+            urlComponents.queryItems = queryItems
+            urlWithToken = urlComponents.url!
+        }
+        return urlWithToken
+    }
+
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+
+        try container.encode(id, forKey: .id)
+        try container.encode(title, forKey: .title)
+        try container.encode(videoDescription, forKey: .description)
```

Comment:
If I understand correctly, we explicitly handle this encoding due to the `videoDescription` property name conflict. The good news is I _believe_ this can be avoided by adding `videoDescription = "description"` to our `CodingKeys`.

This should allow auto syntheses of the `Encodable` protocol and would remove the need for this function.

Reply:
Ah, good to know. I'll try this option and update it accordingly if it works. Thanks for the tip!

Reply:
I tried this and worked as expected 🎊 , I applied this suggestion in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/581/commits/b901dc8f0ae251f2698a1bfa2e7c2a1c4599e6c1. Thanks!